src/command_modules/azure-cli-role/azure/cli/command_modules/role/_validators.py

```diff
@@ -85,6 +85,18 @@ def validate_cert(namespace):
         namespace.cert = x509
 
 
+def process_assignment_namespace(cmd, namespace):
+    from azure.cli.core.util import ConfiguredDefaultSetter
+    if namespace.resource_group_name and namespace.scope:
+        with ConfiguredDefaultSetter(cmd.cli_ctx.config, 'local'):
+            defaults_result = cmd.cli_ctx.config.items(cmd.cli_ctx.config.defaults_section_name)
+        if next((x for x in defaults_result if x.get('value') == namespace.resource_group_name and
```

Comment:
The much more concise way to do this would be:
```Python
resource_group = namespace.resource_group_name
if resource_group and not getattr(resource_group, 'is_default', None):
 …
```

Any values that are those values solely because of a default will have this 'is_default' property.